### PR TITLE
Only consider production deployments

### PR DESCRIPTION
**Pull request by bratseth**
When reasoning about versions, we should not take deployments to "dev"
and "perf" into account.

@mpolden or @jonmv please review

### CODE REVIEW

**Comment by jonmv** [comment]
All changes look good. I didn't look for whether there should be more changes than this.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java [hunk 1]
```diff
@@ -93,7 +95,17 @@ private Application(ApplicationId id, DeploymentSpec deploymentSpec, ValidationO
...
     public Map<Zone, Deployment> deployments() { return deployments; }
-    
+
+    /** 
+     * Returns an immutable map of the current *production* deployments of this
+     * (deployments also includes manually deployed environments)
+     */
+    public Map<Zone, Deployment> productionDeployments() {
+        return deployments.values().stream()
+                                  .filter(deployment -> deployment.zone().environment() == Environment.prod)
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java]: Alignment is off.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java [hunk 2]
```diff
@@ -482,7 +481,7 @@ public Application deleteApplication(ApplicationId id, Optional<NToken> token) {
         try (Lock lock = lock(id)) {
             Optional<Application> application = get(id);
             if ( ! application.isPresent()) return null;
-            if ( ! application.get().deployments().isEmpty())
+            if ( ! application.get().productionDeployments().isEmpty())
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Won't this allow application data in the controller to be deleted while there are still active dev or perf deployments? And potentially make config servers and controller out of sync?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java (hunk 1 above)
Comment (bratseth) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java]: Fixed. Will be in whatever I do next.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java (hunk 2 above)
Comment (bratseth) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: It is ok to delete them, but I see the deployments are not deleted here. Nice catch, Incoming PR then.

Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Deleting and deactivating a deployment is separate code path / API call I think.

Comment (bratseth) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Yes. For non-prod a recursive delete would be nice, but probably not worth the bother.

**Pull request merged**